test/signalrclient-e2e-tests/hub_connection_tests.cpp

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include "stdafx.h"
+#include <string>
+#include "cpprest\details\basic_types.h"
+#include "cpprest\json.h" 
+#include "connection.h"
+#include "hub_connection.h"
+#include "console_log_writer.h"
+
+using namespace signalr;
+using namespace utility;
+using namespace std;
+
+TEST(hub_connection_tests, send_message)
+{
+	shared_ptr<log_writer> Log_writer(make_shared<console_log_writer>());
+
+	hub_connection hubConn(L"http://localhost:8081/SignalR", L"", trace_level::all, Log_writer);
+	string_t message;
+	pplx::event test_event{};
+
+	hub_proxy hubProxy = hubConn.create_hub_proxy(L"hubConnectionAPI");
+
+	hubProxy.on(L"displayMessage", [&message](const web::json::value& arguments)
+	{
+		message = arguments.serialize();
+		std::wcout << "Received message: " << message << endl;
+	});
+
+	hubConn.start().then([&hubConn, &hubProxy, &test_event](pplx::task<void> start_task)
+	{
+		start_task.get();
+		ASSERT_TRUE(hubConn.get_connection_state() == connection_state::connected);
+
+		web::json::value obj{};
+		obj[0] = web::json::value(L"test");
+
+		hubProxy.invoke<web::json::value>(L"displayMessageAll", obj)
+			.then([&test_event](pplx::task<web::json::value> send_task)
+		{
+			auto test = send_task.get();
+			ASSERT_TRUE(test.serialize().find(L"test") != string::npos);
+			test_event.set();
+		});
+	});
+
+	ASSERT_FALSE(test_event.wait(15000));
+
+	std::wcout << "message: " << message << endl;
+
+	ASSERT_TRUE(message.find(L"Broadcast: test") != string::npos);
```

Comment:
In general I think stopping the connection is a wise thing. You capture by `&` a lot and if you get anything from the server when the connection goes out of scope but before it is destroyed you will see random crashes. And it is likely that this happens because a new test might start running and sending messages before the connection is internally cleaned up. Note that internally the `connection_impl` derives from `enable_shared_from_this` and it can easily outlive the `connection`.